hail/python/hailtop/batch/resource.py

```diff
@@ -167,10 +132,48 @@ class JobResourceFile(ResourceFile):
     to be saved.
     """
 
+    def __init__(self, value):
+        super().__init__(value)
+        self._has_extension = False
+
     def _get_path(self, directory):
         assert self._source is not None
         assert self._value is not None
-        return shq(directory + '/' + self._source._uid + '/' + self._value)
+        return f'{directory}/{self._source._job_id}/{self._value}'
+
+    def add_extension(self, extension):
```

Comment:
in other places (e.g. `command`) we don't add a verb, we just state what's being changed. let's stick to that system.

Reply:
That would be a breaking change. Let's do it in another PR.